smash/fcore/operator/md_neural_network.f90

```diff
@@ -0,0 +1,68 @@
+!%      (MD) Module Differentiated.
+!%
+!%      Subroutine
+!%      ----------
+!%
+!%      - forward_mlp
+
+module md_neural_network
+
+    use md_constant !% only : sp
+    use md_algebra !% only: dot_product_2d_1d
+
+    implicit none
+
+contains
+
+    subroutine forward_mlp(weight_1, bias_1, weight_2, bias_2, weight_3, bias_3, input_layer, output_layer)
+        !% The forward pass of the multilayer perceptron used in hydrological model structure
```

Comment:
add a short comment on why hardcoded structure vs // and AD constrains... perhaps in header and doc also